### PR TITLE
Set gradle / maven max heap size to 2048m if not set by the user

### PR DESCRIPTION
**Pull request by bgavrilMS**
The root cause for this issue is the Maven / Gradle max heap size, which by default is set too low. SonarQube uses quite a bit of memory to do its analysis, but this issue isn’t particular to sonar. Large vanilla builds / other plugins would encounter the same problem. 

Setting an env variable named GRADLE_OPTS / MAVEN_OPTS is a way to control the heap size.

### CODE REVIEW

File: Tasks/Gradle/gradletask.ts [hunk 1]
```diff
@@ -124,6 +126,19 @@ function processCodeAnalysisResults(): Q.Promise<void> {
     return sqGradle.uploadSonarQubeBuildSummaryIfEnabled();
 }
 
+// Configure the maximum memory available to Gradle if the user hasn't done so, otherwise it may run ouf of heap space
+function configureMemoryLimits() {
+    if (!process.env['GRADLE_OPTS']) {
+
+        let defaultMavenOpts = '-Xmx4096m';
```
Comment (bgavrilMS) [inline Tasks/Gradle/gradletask.ts]: 4096 might be too much, 2048 might be a better value. When using SonarQube for msbuild we set it to 1024m and we had only 1 customer complain about out of memory exception.

File: Tasks/Gradle/gradletask.ts [hunk 2]
```diff
@@ -124,6 +126,19 @@ function processCodeAnalysisResults(): Q.Promise<void> {
     return sqGradle.uploadSonarQubeBuildSummaryIfEnabled();
 }
 
+// Configure the maximum memory available to Gradle if the user hasn't done so, otherwise it may run ouf of heap space
+function configureMemoryLimits() {
+    if (!process.env['GRADLE_OPTS']) {
```
Comment (bgavrilMS) [inline Tasks/Gradle/gradletask.ts]: The only possible problem I see with this is if the user can configure the heap size differently (e.g. via the build.gradle file) and we would override it. I couldn't find a way to do this though.

**Comment by jpricketMSFT** [comment]
Looks good to me.

**Comment by yacaovsnc** [comment]
Personally, I believe we should expose this option from UI in the "Advanced" section, and default the value to something big enough to run SonarQube.  Hardcoding this number in code hides it from the user, and as a dev I can't simply replicate the build by looking at the build definition anymore.  If we are only a tool runner, I prefer to be as transparent as possible.

That said, if we want to go with this approach, 4096m is too big.   32Bit JRE will fail to start with this number.

File: Tasks/Gradle/gradletask.ts (hunk 1 above)
Comment (yacaovsnc) [inline Tasks/Gradle/gradletask.ts]: Your var name is defaultMavenOpts in gradle file.

**Comment by yacaovsnc** [comment]
LGTM

**Pull request merged**